bfrec

```diff
@@ -146,6 +146,61 @@ fi
 bfb_location=/lib/firmware/mellanox/boot
 capsule_location=/lib/firmware/mellanox/boot/capsule
 
+parse_and_set_capsule_versions()
+{
+    local capsule_file="$1"
+
+    if [ -z "$capsule_file" ]; then
+        echo "ERROR: No capsule file specified"
+        exit 2
+    fi
+
+    if [ ! -f "$capsule_file" ]; then
+        echo "ERROR: Capsule file '$capsule_file' not found"
+        exit 2
+    fi
+
+    # Check if bfver command is available
+    if ! command -v bfver >/dev/null 2>&1; then
+        echo "Boot image parsing: bfver command not found, skipping"
+        return 0
+    fi
+
+    # Use bfver to parse the ATF and UEFI version from the capsule file.
+    BFVER_OUTPUT=$(bfver -f "$capsule_file")
+
+    local atf_version=""
+    local uefi_version=""
+    while IFS= read -r line; do
+        if [[ $line == *"BlueField ATF version:"* ]]; then
+            atf_version=$(echo "$line" | cut -d':' -f3 | xargs)
+        elif [[ $line == *"BlueField UEFI version:"* ]]; then
+            uefi_version=$(echo "$line" | cut -d':' -f2 | xargs)
+        fi
+    done <<< "$BFVER_OUTPUT"
+
+    # Check if both versions were successfully parsed
+    if [ -n "$atf_version" ] && [ -n "$uefi_version" ]; then
+        echo "Boot image detected: ATF pending version: $atf_version, UEFI pending version: $uefi_version"
+
+        # Check if bfver command is available
+        if ! command -v bfcfg >/dev/null 2>&1; then
+            echo "Boot image parsing: bfcfg command not found, skipping"
+            return 0
+        fi
+
+        # Transfer the ATF and UEFI pending version to UEFI.
+        bfcfg --capatfver "$atf_version"
+        bfcfg --capuefiver "$uefi_version"
+
+        echo "Boot image updated to UEFI: ATF pending version: $atf_version, UEFI pending version: $uefi_version"
+    else
+        echo "Boot image parsing: '$capsule_file' does not contain boot image"
```

Comment:
I am not a big fan of this log message. This will rise questions from QA and/or customer.

Again `bfrec` command is meant to initiate capsule updates. Capsules are generic files and not tight to boot image updates only. This message would imply that capsules should contain boot images. Which is not always true.

Anyway, I'll leave it up to you; whether you would like to keep it or omit it.